docs/codeql/codeql-language-guides/basic-query-for-cpp-code.rst

```diff
@@ -14,62 +16,65 @@ The query we're going to run performs a basic search of the code for ``if`` stat
 
-#. Click the project in the search results.
+Before you start writing queries, you need a CodeQL database to run them against. The simplest way to do this is to download a database directly from GitHub.com.
 
-#. Click **Query this project**.
+#. In Visual Studio Code, click the **QL** icon |codeql-ext-icon| in the left sidebar to display the CodeQL extension. 
 
-   This opens the query console. (For information about using this, see `Using the query console <https://lgtm.com/help/lgtm/using-query-console>`__.)
+#. Click **From GitHub** or the GitHub logo |github-db| at the top of the CodeQL extension to open an entry field.
 
-   .. pull-quote::
+#. Copy the URL for the repository into the field and press the keyboard **Enter** key.
 
-      Note
+#. Optionally, if the repository has more than one CodeQL database available, choose which language to download.
 
-      Alternatively, you can go straight to the query console by clicking **Query console** (at the top of any page), selecting **C/C++** from the **Language** drop-down list, then choosing one or more projects to query from those displayed in the **Project** drop-down list.
-
-#. Copy the following query into the text box in the query console:
-
-   .. code-block:: ql
+Information about the progress of the database download is shown in the bottom right corner of Visual Studio Code. When the download is complete, the database is shown with a check mark in the **Databases** section of the CodeQL extension.
 
-      import cpp
+.. image:: ../images/codeql-for-visual-studio-code/database-selected.png
+       :align: center
+       :width: 500
 
-      from IfStmt ifstmt, BlockStmt block
-      where ifstmt.getThen() = block and
-        block.getNumStmt() = 0
-      select ifstmt, "This 'if' statement is redundant."
+Running the query
+-----------------
 
-   LGTM checks whether your query compiles and, if all is well, the **Run** button changes to green to indicate that you can go ahead and run the query.
+#. In Visual Studio Code, create a new folder to store your experimental queries for C and C++ CodeQL databases. For example, ``cpp-experiments``.
 
-#. Click **Run**.
+#. Create a ``qlpack.yml`` file in your experiments folder with the contents shown below. This tells the CodeQL extension that any queries you create in the folder are intended to run on C or C++ CodeQL databases.
 
-   The name of the project you are querying, and the ID of the most recently analyzed commit to the project, are listed below the query box. To the right of this is an icon that indicates the progress of the query operation:
+   .. code-block:: yaml
 
-   .. image:: ../images/query-progress.png
-       :align: center
+      name: github-owner/cpp-experiments
+      version: 0.0.1
+      dependencies:
+         codeql/cpp-all: ^0.1.2
 
-   .. pull-quote::
+#. Create a second new file in your experiements folder with the ``.ql`` file extension. You will write your first query in this file.
 
```

Comment:
For simplicity, we could point users to the quick query command here: https://codeql.github.com/docs/codeql-for-visual-studio-code/analyzing-your-projects/#running-a-quick-query

It would save folks having to create a new qlpack.yml file. However, they would have to copy the contents of the synthetic CodeQL pack to somewhere permanent if they wanted to retain their test queries, but that could be an optional extra step at the end of this procedure (and we could simply link to the docs).